zenoh-plugin-ros2dds/src/ros2_utils.rs

```diff
@@ -228,16 +228,22 @@ impl CddsRequestHeader {
         buf[16] = self.is_little_endian as u8;
 
         hashmap.insert(ATTACHMENT_KEY_REQUEST_HEADER, buf);
-        ZBytes::from_iter(hashmap.iter())
+
+        let mut writer = ZBytes::writer();
+        for (k, v) in hashmap.iter() {
+            writer.append(ZBytes::from(k));
+            writer.append(ZBytes::from(v));
+        }
+        writer.finish()
     }
 }
 
 impl TryFrom<&ZBytes> for CddsRequestHeader {
     type Error = ZError;
 
     fn try_from(value: &ZBytes) -> Result<Self, Self::Error> {
-        let hashmap: HashMap<[u8; 3], [u8; 17]> =
```

Comment:
On receiving side, we expect the the attachment `value` to contain only:
- the `ATTACHMENT_KEY_REQUEST_HEADER`
- the header (16 bytes)

Rather than using a hashmap, we could just do:
- try to read `len(ATTACHMENT_KEY_REQUEST_HEADER)` bytes (return `Err` if fail)
- check if those bytes are equal to `ATTACHMENT_KEY_REQUEST_HEADER` (return `Err` if fail)
- try to read 16 bytes (return `Err` if fail)

Reply:
Reading Bytes from slice failing if not enough bytes are found, or if bytes do not match `ATTACHMENT_KEY_REQUEST_HEADER` values.